### ISSUE TITLE
Disabled items breaking focus in menu

### ISSUE THREAD

**Issue by roginfarrer** [paypal/downshift#557]
- `downshift` version: 2.1.1
- `node` version: 8.9.4
- `npm` (or `yarn`) version: 6.1.0

![Example of the issue](https://dzwonsemrish7.cloudfront.net/items/3P2v1H2k2M1e0O2D0D3S/Screen%20Recording%202018-08-20%20at%2004.43%20PM.gif?v=9c43f8b3)

**Problem description**:

I have a Downshift dropdown that has a menu stay open after an option is selected. Selected or disabled options also remain in the menu. When clicking on one of those selected or disabled options, the focus goes to the container, which messes up the highlightedIndex and prevents the user from selecting an option again.

**Suggested solution**:

I think the key piece we're missing here is this event handler that's omitted when an option is disabled:

https://github.com/paypal/downshift/blob/9e3bfbf6cabf7ec92d1d3eee1929ef789eda58ad/src/downshift.js#L840-L845

I think we can safely add this event handler to the disabled eventHandlers:

```js
const eventHandlers = rest.disabled ? {onMouseDown: enabledEventHandlers.onMouseDown} : enabledEventHandlers
```

**Comment by roginfarrer** [comment]
I opened a pull request, let me know what your thoughts!

**Comment by roginfarrer** [comment]
Resolved by https://github.com/paypal/downshift/pull/558, released in v2.1.4
